config/mozzi_config_documentation.h

```diff
@@ -0,0 +1,258 @@
+*/
+
+/** @ingroup config
+ * @def MOZZI_COMPATIBILITY_LEVEL
+ *
+ * Mozzi generally tries to keep your old sketches working, but we continue to find new (and hopefully better) approaches to old problems.
+ * Sometimes, keeping API compatibilty with the pre-existing solution may come with a smaller or larger penalty in terms of performance or code size.
+ * Therefore - if your sketch supports it - you may be able to get some minor benefit from disabling compatibility code.
+ *
+ * Currently supported values are:
+ *   - MOZZI_COMPATIBILITY_1_1 - try to support sketches written for Mozzi version 1.1 (or possibly lower); this is the default when including MozziGuts.h
+ *   - MOZZI_COMPATIBILITY_2_0 - try to support sketches written for Mozzi version 2.0; this is - currently - the default when including Mozzi.h
+ *   - MOZZI_COMPATIBILITY_LATEST - always live on the bleeding edge
+ *
+ * @note
+ * MOZZI_COMPATIBILITY_V1_1 does not guarantee, that *everything* from Mozzi 1.1 will continue to work, just that we're doing a reasonable effort.
+*/
+
+/** @ingroup config
+ * @def MOZZI_AUDIO_MODE
+ *
+ * @brief Configure how Mozzi outputs generated sounds.
+ *
+ * @note
+ * Not all options are available on all platforms, and several options require specific wiring or external components to function on top of this!
+ * When customizing this, it is highly recommended to start experimenting with a simple and known-to-work sketch (such as a basic sinewave) to verify that your
+ * hardware setup is correct. Similarly, if you observe problems running your "real" sketch, it is often a good idea ot test your sketch with the default audio mode,
+ * too (by leaving this option, and preferrably all others, unset).
+ *
+ * Refer to the @ref hardware specific documentation for which modes are supported on your hardware, and further details!
+ *
+ * Supported values:
+ *   - MOZZI_OUTPUT_PWM Output using pulse width modulation (PWM) on a GPIO pin. This is the default on most platforms.
+ *                      On the Arduino Uno (more generally ATMEGA328P), this allows for a sample resolution of 488 (almost 9 bits) on pin 9.
+ *                      Usable pins and resolution will be different on other boards.
+ *   - MOZZI_OUTPUT_2PIN_PWM Output using pulse width modulation on two GPIO pins, where one pin represents the lower bits, and the other the higer bits of the sample.
+ *                      On the Aduino Uno, this allows for 14 bits of resolution on pins 9 (low) and 10 (high). For further information (wiring etc.) see @ref hardware_avr_2pin.
+ *   - MOZZI_OUTPUT_EXTERNAL_TIMED Output is not controlled by Mozzi itself, but left to the user sketch. This setting allows to completely customize the audio output, e.g.
+ *                      for connecting to external DACs. For more detail, see @ref external_audio
+ *   - MOZZI_OUTPUT_EXTERNAL_CUSTOM As above, but additionally bypassing Mozzi's sample buffer. For more detail, see @ref external_audio
```

Comment:
I understand it makes sense to have an `EXTERNAL_CUSTOM` with respect to the fact that we sometimes bypass the buffer for some output options (directly in Mozzi) but, out of curiosity, does such a thing as ever been employed in an external fashion?

Reply:
Maybe not thus far, but it seems to make sense in that:
- it's totally easy to implement ;-)
- when interfacing with third-party output libs / mechanisms, it seems quite likely that some from of buffering / rate control is already in place, there.